node/pkg/watchers/evm/ccq_backfill.go

```diff
@@ -0,0 +1,303 @@
+package evm
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/certusone/wormhole/node/pkg/watchers/evm/connectors"
+
+	"go.uber.org/zap"
+
+	ethHexUtil "github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/rpc"
+	ethRpc "github.com/ethereum/go-ethereum/rpc"
+)
+
+const CCQ_MAX_BATCH_SIZE = int64(1000)
+const CCQ_TIMESTAMP_RANGE_IN_MS = uint64(30 * 60)
+
+type ccqBackfillRequest struct {
+	timestamp uint64
+}
+
+// ccqRequestBackfill submits a request to backfill a gap in the timestamp cache.
+func (w *Watcher) ccqRequestBackfill(timestamp uint64) {
+	select {
+	case w.ccqBackfillChannel <- &ccqBackfillRequest{timestamp: timestamp}:
+		w.ccqLogger.Debug("published backfill request", zap.Uint64("timestamp", timestamp))
+	default:
+		// This will get retried next interval.
+		w.ccqLogger.Error("failed to published query response error to handler", zap.Uint64("timestamp", timestamp))
+	}
+}
+
+// ccqBackfiller is the runnable that handles backfilling the timestamp cache. It initializes the cache and then processes incoming backfill requests from CCQ.
+func (w *Watcher) ccqBackfiller(ctx context.Context) error {
+	if err := w.ccqBackfillInit(ctx); err != nil {
+		return fmt.Errorf("failed to backfill timestamp cache: %w", err)
+	}
+
+	for {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		case evt := <-w.ccqBackfillChannel:
+			w.ccqPerformBackfill(ctx, evt)
+		}
```

Comment:
Do you need a default case?

Reply:
I don't think so. We want it to just block if there's nothing to do.